pattern/BUILD

```diff
@@ -56,6 +56,7 @@ native_java_libraries(
     native_libraries_deps = [
         # Internal dependencies
         "//common:common",
+        "//encoding:encoding",
```

Comment:
this was the main new depedency we wanted to introduce, in order to allow `ValueConstraint` to do comparisions (to check whether two `ValueConstraints` contradict each other). Now that comparators for each value type live in `//encoding`, we can introduce this dependency without introduce a `//graph` dependency.